### ISSUE TITLE
Upgrading from 13.0.2 to 13.1.1 breaks configuration binding in project

### ISSUE THREAD

**Issue by skttl** [umbraco/Umbraco.UIBuilder.Issues#96]
**Describe the bug**
Don't really know how, but here goes.

We have a project with UI Builder. In that project we add some configuration through appsettings.json like
```json
{
  "Umbraco": {
    "Community": {
      "ImageSharpRemoteImages": {
        "Settings": [
          {
            "Prefix": "/remote",
            "AllowedDomains": [ "*" ],
            "MaxBytes":  24
          }
        ]
      }
    },
}
```
MaxBytes is an int property in the IOptions model.

Before upgrading UI Builder this would result in IOptions<RemoteImageProviderOptions> having a correct value of 24 in the MaxBytes.

After upgrading, the int value from appsettings.json fails to bind, resulting in the value falling back to the default (eg. 0). This happens for all int values in IOptions models. 

The above example is using a package - so you can see how the [model is bound in the source code](https://github.com/skttl/ImageSharpCommunity.Providers.Remote/blob/main/src/Umbraco.Community.ImageSharpRemoteImages/UmbracoBuilderExtensions.cs).

I've also tried with other config settings using `builder.Services.Configure<MyOptions>(builder.Config.GetSection("Path:To:MyOptions"))` with the same error.

Has anything changed between 13.0 and 13.1 that could cause this?

**Comment by skttl** [comment]
Another example, the configured port for smtp settings eg.

```json
  "Umbraco": {
    "CMS": {
      "Content": {
        "MacroErrors": "Throw"
      },
      "Examine": {
        "LuceneDirectoryFactory": "Default"
      },
      "Global": {
        "MainDomLock": "",
        "Smtp": {
          "From": "it@ecreo.dk",
          "Host": "localhost",
          "Username": null,
          "Password": null,
          "Port": 25
        }
      },
```

Turns up as 0 (screenshot from Diplo God Mode)
![image](https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/3726467/5b6edbc2-8bf3-40cd-9f06-38250057c767)

**Comment by skttl** [comment]
Just tried out the same in a shiny new blank install

Installed using this script (Diplo.GodMode is added, so the config values can be read through the backoffice)
```
# Ensure we have the version specific Umbraco templates
dotnet new install Umbraco.Templates::13.2.2 --force

dotnet new umbraco --force -n "13-2-2-ui-builder-config-bug" --friendly-name "Administrator" --email "admin@admin.com" --password "admin@admin.com" --development-database-type SQLite

#Add Packages
dotnet add "13-2-2-ui-builder-config-bug" package Diplo.GodMode
dotnet add "13-2-2-ui-builder-config-bug" package Umbraco.UIBuilder

dotnet run --project "13-2-2-ui-builder-config-bug"
#Running
```

Add some smtp settings in appsettings.json

```json
...
  "Umbraco": {
    "CMS": {
      "Global": {
        "Id": "c0f6a079-3881-47ad-83d9-cdbdc660e2af",
        "SanitizeTinyMce": true,
        "Smtp": {
          "From": "admin@admin.com",
          "Port": 1322
        }
      },
...
```

Open up Umbraco, go to Settings, God Mode, Diagnostics. Find the smtp settings:
![image](https://github.com/umbraco/Umbraco.UIBuilder.Issues/assets/3726467/b9b67cf7-a548-43bd-9ac0-aabf96463b70)


As you can see, port is 0 here - but should have been 1322 according to appsettings.json.

**Comment by acoumb** [comment]
Hi @skttl ,

The issue you are experiencing is a result of a custom `Udi` type converter I've introduced with latest releases. 

I am looking into this to work on a fix.

Thanks,
Adrian

**Comment by patlit** [comment]
Having same issues after upgrading to 13.1.1 to day. All IOptions of type integer now returns 0
@acoumb Any idea when this will be fixed? Its causing issues for SMTP and image cache configuration and several of our IOptions usages. Do we need to downgrade UI builder? 
Thanks,
Patrik

**Comment by acoumb** [comment]
Hi @skttl , @patlit ,

The issue has been fixed with the latest minor release [13.1.2](https://www.nuget.org/packages/Umbraco.UIBuilder/13.1.2).

Thank you,
Adrian
